### PR TITLE
kubectl create cluster-scope resources special --namespace should be …

### LINKED ISSUE #361: kubectl create clusterrolebinding: must error if --namespace is specified
Users can easily confuse

```
kubectl create rolebinding serviceaccounts-view \
  --clusterrole=view \
  --group=system:serviceaccounts:my-namespace \
  --namespace=my-namespace
```

and

```
kubectl create clusterrolebinding serviceaccounts-view \
  --clusterrole=view \
  --group=system:serviceaccounts:my-namespace \
  --namespace=my-namespace
```

In latter command, the `--namespace` argument is SILENTLY IGNORED because clusterrolebinding is not a namespaced resource.

In such cases (silently ignoring something) we should be explicit and if `--namespace` is specified, `kubectl create clusterrolebinding` should error out!

### PR DESCRIPTION
**Pull request by zjj2wry**
…failed

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes/kubectl/issues/361

**Special notes for your reviewer**:
/kind bug
/sig cli
/cc @liggitt @ahmetb @juanvallejo @soltysh 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
@kubernetes/sig-cli-pr-reviews 
thoughts on tightening validation here?

**Comment by liggitt** [comment]
I'm unsure about making this behave differently than `kubectl create -f … -n ...` which doesn't care if you specify a namespace with a file containing cluster scoped resources

File: pkg/kubectl/cmd/create.go [hunk 1]
```diff
@@ -343,7 +343,11 @@ func RunCreateSubcommand(f cmdutil.Factory, cmd *cobra.Command, out io.Writer, o
 	obj = info.Object
 
 	if !options.DryRun {
-		obj, err = resource.NewHelper(client, mapping).Create(namespace, false, info.Object)
+		help := resource.NewHelper(client, mapping)
+		if !help.NamespaceScoped && nsOverriden {
+			return fmt.Errorf("%s is cluster-scope resources, can not special --namespace %s", gvk.Kind, namespace)
```
Comment (ahmetb) [inline pkg/kubectl/cmd/create.go]: > %s is cluster-scope resources, can not special --namespace %s

unfortunately this is not a good error message grammatically.

Please consider:

> %s is not a namespaced resource, --namespace should not be specified

**Comment by ahmetb** [comment]
> I'm unsure about making this behave differently

Reasoning provided at the linked issue. I think `-f` is a different case (as it can contain multiple resources, some namespaced and some don't).

But in `kubectl [create,delete,edit]` (without `-f`) case we have a clear way of determining if a command is explicitly specifying `--namespace` but the resource kind is non-namespaced.

**Comment by juanvallejo** [comment]
Do we have this kind of validation elsewhere in the code?
I would be more in favor of displaying a warning / log message about this at most.

**Comment by zjj2wry** [comment]
> Do we have this kind of validation elsewhere in the code?
no，just add this validation in kubectl create subcommand. now that kubectl get specified namespace will be ignored，I agree to add log message

**Comment by soltysh** [comment]
If the user has the necessary access rights to create namespace- and cluster-scope resources he should know the difference and be able to figure out what will be created. In other cases (iow. no cluster-scope access) the operation will fail either way. I'd unnecessarily complicate the code.
/hold

**Pull request closed without merging**